### PR TITLE
commit-formatter: init at 0.2.1

### PR DESCRIPTION
**Pull request by Eliot00**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Eliot00** [comment]
Want to port to stable release.

File: pkgs/applications/version-management/commit-formatter/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    owner = "Eliot00";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "EYzhb9jJ4MzHxIbaTb1MxeXUgoxTwcnq5JdxAv2uNcA=";
+  };
+
+  cargoSha256 = "AeHQCoP1HOftlOt/Yala3AXocMlwwIXIO2i1AsFSvGQ=";
+
+  meta = with lib; {
+    description = "A CLI tool to help you write git commit ";
```
Comment (bobby285271) [inline pkgs/applications/version-management/commit-formatter/default.nix:17]:
```suggestion
    description = "A CLI tool to help you write git commit";
```

File: pkgs/applications/version-management/commit-formatter/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    homepage = "https://github.com/Eliot00/commit-formatter";
+    license = licenses.asl20;
+    maintainers = with lib.maintainers; [ elliot ];
```
Comment (bobby285271) [inline pkgs/applications/version-management/commit-formatter/default.nix:20]:
```suggestion
    maintainers = with maintainers; [ elliot ];
```

File: pkgs/applications/version-management/commit-formatter/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    rev = "v${version}";
+    sha256 = "EYzhb9jJ4MzHxIbaTb1MxeXUgoxTwcnq5JdxAv2uNcA=";
+  };
+
+  cargoSha256 = "AeHQCoP1HOftlOt/Yala3AXocMlwwIXIO2i1AsFSvGQ=";
+
+  meta = with lib; {
+    description = "A CLI tool to help you write git commit ";
+    homepage = "https://github.com/Eliot00/commit-formatter";
+    license = licenses.asl20;
```
Comment (fabaff) [inline pkgs/applications/version-management/commit-formatter/default.nix:19]: Looks like that it's dual-licensed ([MIT](https://github.com/Eliot00/commit-formatter/blob/main/LICENSE-MIT) and [ASL](https://github.com/Eliot00/commit-formatter/blob/main/LICENSE-APACHE)).

```suggestion
    license = with licenses; [ asl20 /* or */ mit ];
```

**Comment by bobby285271** [comment]
https://github.com/Eliot00/commit-formatter/blob/2752a3b845b3d43a4442f6fe64b619af3f46eafd/src/main.rs#L42
Maybe we need to wrap the program to make sure we have git?

**Comment by lovesegfault** [comment]
> https://github.com/Eliot00/commit-formatter/blob/2752a3b845b3d43a4442f6fe64b619af3f46eafd/src/main.rs#L42 Maybe we need to wrap the program to make sure we have git in PATH?
> 
> ```nix
> wrapProgram $out/bin/git-cf --prefix PATH : "${lib.makeBinPath [ git ]}"
> ```

It's perfectly normal for programs to rely on having certain things in `PATH`. I don't think we should be wrapping anything here.

Imagine if I have `myCustomGit` in path as `git` and for whatever reason this program isn't working as I expect. Finding out someone had wrapped it to use another `git` than the one in my `PATH` for no good reason would be very frustrating.

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 154110` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>commit-formatter</li>
  </ul>
</details>

**Pull request merged**